### ISSUE TITLE
ValueError: Optimizer weight shape (6,) not compatible with provided weight shape (800, 6)

### ISSUE THREAD

**Issue by ShreyaKhare** [nikicc/twitter-emotion-recognition#3]
I get this error when i try to load an existing model.

**Comment by nikicc** [comment]
Can you please paste the whole stack trace? Also are you on Keras version 1?

**Comment by ShreyaKhare** [comment]
Can you specifiy tensorflow and keras version that was used for training

**Comment by nikicc** [comment]
We only used Theano, no Tensorflow. I think most experiments were run on Theano `0.8.2`.

**Comment by ghost** [comment]
I have Keras installed. Can I use Keras here to implement the codes?

**Comment by nikicc** [comment]
> Can I use Keras here to implement the codes?

@HangsunKim not sure I understand the question. You need `keras==1.1.0` and `pandas` installed, as stated in [requirements.txt](https://github.com/nikicc/twitter-emotion-recognition/blob/master/requirements.txt). After that, you should be able to run the code. Check the [example of usage](https://github.com/nikicc/twitter-emotion-recognition#example-usage).

**Comment by ghost** [comment]
@nikicc Thanks for your reply. I have the pandas and keras library installed. I've always done deep learning by importing keras, as tensforflow as the backend. I think then I'll be able to run the codes then. I asked this question because you somewhere indicated that the codes are run on theano. That was my concern.

**Comment by nikicc** [comment]
>  I asked this question because you somewhere indicated that the codes are run on theano.

Yes, the models were trained using Keras with Theano backend. I'm not sure if the models can be loaded and used with Tensorflow for not. Let me know if this will work for you on Tensorflow.

**Comment by ghost** [comment]
@nikicc   I'm trying to run the demo codes with Theano as backend. 
I found a way to change the backend engine easily from Tensorflow to Theano. 
But, when I run the first line of codes.
I get this error, ImportError: No module named html

Do you know what could be wrong and how I could fix this? Thanks in advance.

**Comment by nikicc** [comment]
> I get this error, ImportError: No module named html

Are you sure you're on Python 3? `html` is a standard Python's [library](https://docs.python.org/3/library/html.parser.html).

**Comment by ghost** [comment]
@nikicc   Oh, I was using python 2.7. Okay, so I tried on python 3.6, but now I get the 
error ModuleNotFoundError: No module named 'theano'.
Do I just do "pip install theano" in my command prompt?

**Comment by nikicc** [comment]
> Do I just do "pip install theano" in my command prompt?

To be sure, use the same version of Theano that I did: `pip install theano==0.8.2` should do it.

**Comment by ghost** [comment]
@nikicc Thanks, I installed theano 0.8.2 but now I get the below error. Are you familiar with this error?
. collect2.exe: error: ld returned 1 exit status status=1): C:\Users\Anaconda3\libs/python36.lib: error adding symbols: File in wrong format

**Comment by nikicc** [comment]
@HangsunKim no idea what that error is about, sorry.

**Comment by ghost** [comment]
@nikicc   It's okay.. I'm in Python 3.6. May I know which python version you are using? Thanks.

**Comment by nikicc** [comment]
I think I was on 3.6 as well.

**Comment by ghost** [comment]
I now get this error.
ValueError: Tensor conversion requested dtype int32 for Tensor with dtype float32: 'Tensor("embedding_1/random_uniform:0", shape=(412, 25), dtype=float32)'
I was wondering if you know this error. 
I referred to 
https://github.com/keras-team/keras/issues/7460.
However, there's no clear solution to it..

**Comment by nikicc** [comment]
Don't know. Are you sure you are on `keras==1.1.0`?

**Comment by SidShenoy** [comment]
Hey! @HangsunKim Finally solved this issue. Follow these steps:

1. Only install the packages mentioned in requirements.txt (with the exact same versions) and no other package (like tensorflow). Uninstall tensorflow if already installed.

2. Change the backend of 'keras' from 'tensorflow' to 'theano'. Instructions to do  this have been mentioned on the 'keras' website (https://keras.io/backend/):

![image](https://user-images.githubusercontent.com/18630586/38103745-d8080878-33a4-11e8-8a0b-efbeafaf8b36.png)

Cheers!

**Comment by nikicc** [comment]
@SidShenoy  thanks for posting this!

**Comment by jhaozz** [comment]
I couldn't install the version, any problem? it says 'could not find a version that satisfies the requirement emotion_predictor (from versions: ) No matching distribution found for emotion_predictor.'

**Comment by venkatshan707** [comment]
sir please give where i want to run the following code :  in jupiter notebook or anaconda prompt

**Comment by venkatshan707** [comment]
these codes:
>>> from emotion_predictor import EmotionPredictor

>>> model = EmotionPredictor(classification='ekman', setting='mc')
 
>>> tweets = [
    "Watching the sopranos again from start to finish!",
    "Finding out i have to go to the  dentist tomorrow",
    "I want to go outside and chalk but I have no chalk",
    "I HATE PAPERS AH #AH #HATE",
    "My mom wasn't mad",
    "Do people have no Respect for themselves or you know others peoples homes",
]

**Comment by venkatshan707** [comment]
![screenshot 320](https://user-images.githubusercontent.com/30896136/52804856-4532aa80-30ab-11e9-8093-8e0dd1ea8068.png)
![screenshot 321](https://user-images.githubusercontent.com/30896136/52804858-45cb4100-30ab-11e9-9f4d-5f995c262958.png)

**Comment by venkatshan707** [comment]
i got those errors while running through jupitter notebook

**Comment by nikicc** [comment]
@venkatshan707 seem like the issue is in the backend your Keras is trying to use. According to the above screenshots your Keras wants to invoke a `tensorflow` backend, while you will need to use `theano` backend to use the emotion predictiors from this repository. More precisely, you'll need `theano==0.8.2` (see the [requirements file](https://github.com/nikicc/twitter-emotion-recognition/blob/master/requirements.txt)). Check [this page in the Keras documentation](https://keras.io/backend/) to figure out how to switch to `theano` backend.

**Comment by venkatshan707** [comment]
Sir,after setting a background from where we want to start our code
execution.I am just started to learn(very beginner) data science.I am doing
my final year project with your code. I failed to execute your code. Can
you please,send user manual of that code execution (or) please,teach me how
we want to execute our code in step by step manner using TeamViewer or
anydesk.please I tried my level best.please help me sir.

**Comment by nikicc** [comment]
@venkatshan707 if you set the backend up, then you should be able to start the code either from Jupiter or python script. What is the error you are seeing now after the backend has been set?

Sorry but I don't have any user manual I could share with you.

**Comment by venkatshan707** [comment]
sir ,can you please help me with teamviewer .

**Comment by venkatshan707** [comment]
just 10 mins

**Comment by nikicc** [comment]
@venkatshan707 I think it's easier if we keep the communication here. What is the error you are seeing now?

**Comment by drcs8283** [comment]
how to import emotion predicator module...

**Comment by nikicc** [comment]
@chandrasekhar8283 see https://github.com/nikicc/twitter-emotion-recognition#example-usage

**Comment by Mounika2825** [comment]
how to import emotion predicator module...
i found this error while importing 
ModuleNotFoundError: No module named 'emotion_predictor'

**Comment by nikicc** [comment]
@Mounika2825 are you in the correct folder when executing your code? The [example](https://github.com/nikicc/twitter-emotion-recognition#example-usage) assumes you are in the same folder where the [emotion_predictor.py](https://github.com/nikicc/twitter-emotion-recognition/blob/master/emotion_predictor.py) file is.

**Comment by Mounika2825** [comment]
How to get accuracy by using this code..

**Comment by Mounika2825** [comment]
Thanks a lot.. i executed the code.... Can i get accuracy by using this code In models ?

**Comment by Mounika2825** [comment]
@nikicc Thanks a lot.. i executed the code.... Can i get accuracy by using this code  ?

**Comment by Mounika2825** [comment]
how can i implement this code in google colab?

**Comment by nikicc** [comment]
@Mounika2825 the current code does not support calculating the accuracy. You would need to implement that yourself. Also, not sure about the google colab.

**Comment by Mounika2825** [comment]
How can i get accuracy by using models of unison

**Comment by nikicc** [comment]
@Mounika2825 not sure which accuracy are you interested in. If you want to know the accuracy of the models as we estimated it, look at the [paper](https://ieeexplore.ieee.org/document/8295234) for more details; but approximately the accuracy is around 70%. If you want to know the accuracy of the model on your own data set, then you'd need to calculate that yourself.
